Mathlib/CategoryTheory/ChosenFiniteProducts.lean

```diff
@@ -543,6 +556,247 @@ end ChosenFiniteProductsComparison
+scoped notation "𝟘_" => initialObj
+
+variable {C : Type u} [Category.{v} C] [ChosenFiniteCoproducts C]
+
+/-- The chosen coproduct of two objects -/
+abbrev coprodObj (X Y : C) : C := (coproduct X Y).cocone.pt
+
+/-- Notation for the chosen coproduct of two objects -/
+scoped infixr:70 " ⊕ₒ " => coprodObj
+
+/--
+The unique map from the initial object.
+-/
+def fromZero (X : C) : 𝟘_ C ⟶ X :=
+  initial.isColimit.desc <| .mk _ <| .mk (fun x => x.as.elim) fun x => x.as.elim
+
+instance (X : C) : Unique (𝟘_ C ⟶ X) where
+  default := fromZero _
+  uniq _ := initial.isColimit.hom_ext fun ⟨j⟩ => j.elim
+
+/--
+This lemma follows from the preexisting `Unique` instance, but
+it is often convenient to use it directly as `apply fromZero_unique` forcing
+lean to do the necessary elaboration.
+-/
+lemma fromZero_unique {X : C} (f g : 𝟘_ _ ⟶ X) : f = g :=
+  Subsingleton.elim _ _
+
+/--
+Construct a morphism to the coproduct given its two components.
+-/
+def desc {T X Y : C} (f : X ⟶ T) (g : Y ⟶ T) : X ⊕ₒ Y ⟶ T :=
+  (coproduct X Y).isColimit.desc <| Limits.BinaryCofan.mk f g
+
+/--
+The first injecton into the coproduct
+-/
+def inl (X Y : C) : X ⟶ X ⊕ₒ Y :=
+  letI F : Limits.BinaryCofan X Y := (coproduct X Y).cocone
+  F.inl
+
+/--
+The second injection into the coproduct
+-/
+def inr (X Y : C) : Y ⟶ X ⊕ₒ Y :=
+  letI F : Limits.BinaryCofan X Y := (coproduct X Y).cocone
+  F.inr
+
+@[simp, reassoc]
+lemma inl_desc {T X Y : C} (f : X ⟶ T) (g : Y ⟶ T) : inl X Y ≫ desc f g = f := by
+  simp [inl, desc]
+
+@[simp, reassoc]
+lemma inr_desc {T X Y : C} (f : X ⟶ T) (g : Y ⟶ T) : inr X Y ≫ desc f g = g := by
+  simp [inr, desc]
+
+instance epi_desc_of_epi_left {T X Y : C} (f : X ⟶ T) (g : Y ⟶ T)
+    [Epi f] : Epi (desc f g) :=
+  epi_of_epi_fac <| inl_desc _ _
+
+instance epi_desc_of_epi_right {T X Y : C} (f : X ⟶ T) (g : Y ⟶ T)
+    [Epi g] : Epi (desc f g) :=
+  epi_of_epi_fac <| inr_desc _ _
+
+@[ext 1050]
+lemma hom_ext {T X Y : C} (f g : X ⊕ₒ Y ⟶ T)
+    (h_inl : inl X Y ≫ f = inl X Y ≫ g)
+    (h_inr : inr X Y ≫ f = inr X Y ≫ g) :
+    f = g :=
+  (coproduct X Y).isColimit.hom_ext fun ⟨j⟩ => j.recOn h_inl h_inr
+
+-- Similarly to `CategoryTheory.Limits.coprod.desc_comp`, we do not make the `assoc` version a simp
+-- lemma
+@[reassoc, simp]
+lemma desc_comp {V W X Y : C} (f : V ⟶ W) (g : X ⟶ V) (h : Y ⟶ V) :
+    desc g h ≫ f = desc (g ≫ f) (h ≫ f) := by ext <;> simp [inl_desc_assoc, inr_desc_assoc]
+
+@[simp]
+lemma desc_inl_inr {X Y : C} : desc (inl X Y) (inr X Y) = 𝟙 (X ⊕ₒ Y) := by ext <;> simp
+
+@[simp]
+lemma desc_inl_comp_inr_comp {X Y Z : C} (f : Y ⊕ₒ Z ⟶ X) :
+    desc (inl Y Z ≫ f) (inr Y Z ≫ f) = f := by ext <;> simp
+
+-- TODO: if `AddMonoidalCategory` is added, switch to using `addHom` from there, analogously to how
+-- `ChosenFiniteProducts` uses `tensorHom` from `MonoidalCategory`
+
+/-- The coproduct of two morphisms -/
+abbrev addHom {X Y X' Y' : C} (f : X ⟶ Y) (g : X' ⟶ Y') : X ⊕ₒ X' ⟶ Y ⊕ₒ Y' :=
```

Comment:
Please make this into a `def` and rewrite backwards with `desc_comp_inl_inr` as required. In the past we have used way too many abbreviations, leading to APIs which are very difficult to understand as many applicable lemmas talk about something completely different than what you see in the goal. It is much better to use `def` to create an abstraction boundary and rephrase the relevant API using the new name (which you are already doing below).